x/evm/types/utils.go

```diff
@@ -4,26 +4,33 @@
 package types
 
 import (
+	"bytes"
 	"fmt"
 	"math/big"
 
-	"github.com/cosmos/gogoproto/proto"
-
 	errorsmod "cosmossdk.io/errors"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
-
+	"github.com/cosmos/gogoproto/proto"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/math"
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
-// DefaultPriorityReduction is the default amount of price values required for 1 unit of priority.
-// Because priority is `int64` while price is `big.Int`, it's necessary to scale down the range to keep it more pratical.
-// The default value is the same as the `sdk.DefaultPowerReduction`.
-var DefaultPriorityReduction = sdk.DefaultPowerReduction
+var (
+	// DefaultPriorityReduction is the default amount of price values required for 1 unit of priority.
+	// Because priority is `int64` while price is `big.Int`, it's necessary to scale down the range to keep it more pratical.
+	// The default value is the same as the `sdk.DefaultPowerReduction`.
+	DefaultPriorityReduction = sdk.DefaultPowerReduction
 
-var EmptyCodeHash = crypto.Keccak256(nil)
+	// EmptyCodeHash is keccak256 hash of nil to represent empty code.
+	EmptyCodeHash = crypto.Keccak256(nil)
+)
+
+// BytesAreEmptyCodeHash checks if the given byte slice represents an empty code hash.
+func BytesAreEmptyCodeHash(bz []byte) bool {
+	return bytes.Equal(bz, EmptyCodeHash)
```

Comment:
```suggestion
	return bytes.Equal(bz, EmptyCodeHash) || bytes.Equal(bz, common.Hash{}.Bytes())
```

Reply:
This is already implemented [here](https://github.com/evmos/evmos/blob/dc3b28d8bd000b72c9483acc051cf74e43e8b043/types/validation.go#L13-L16) and used for a different purpose - all checks regarding the code hash are against `keccak256(nil)`, all checks against `common.Hash{}` are related to logs.

I personally think the separated methods are cleaner here, do you insist on merging them?